### PR TITLE
Use HTTP auth in blackbox exporter 

### PR DESCRIPTION
**Pull request by Conorbro**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -54,17 +54,17 @@ type Module struct {
...
+	PreferredIPProtocol    string                  `yaml:"preferred_ip_protocol"`
+	NoFollowRedirects      bool                    `yaml:"no_follow_redirects"`
+	FailIfSSL              bool                    `yaml:"fail_if_ssl"`
+	FailIfNotSSL           bool                    `yaml:"fail_if_not_ssl"`
+	Method                 string                  `yaml:"method"`
+	Headers                map[string]string       `yaml:"headers"`
+	FailIfMatchesRegexp    []string                `yaml:"fail_if_matches_regexp"`
+	FailIfNotMatchesRegexp []string                `yaml:"fail_if_not_matches_regexp"`
+	Body                   string                  `yaml:"body"`
+	HTTPClientConfig       config.HTTPClientConfig `yaml:"http_client_config"`
```
Comment (brian-brazil) [inline main.go]: this should be marked inline

File: http.go [hunk 2]
```diff
@@ -120,20 +120,23 @@ func probeHTTP(target string, module Module, registry *prometheus.Registry) (suc
...
 
-	tlsconfig, err := config.NewTLSConfig(&module.HTTP.TLSConfig)
+	client, err := config.NewHTTPClientFromConfig(httpClientConfig)
 	if err != nil {
-		log.Errorf("Error generating TLS config: %s", err)
-		return false
+		log.Errorf("Error generating HTTP client: %v", err)
 	}
+	client.Timeout = module.Timeout
+	tlsConfig, err := config.NewTLSConfig(&httpClientConfig.TLSConfig)
```
Comment (brian-brazil) [inline http.go]: NewHTTPClientFromConfig handles this, no need to redo it

File: http.go [hunk 3]
```diff
@@ -120,20 +120,23 @@ func probeHTTP(target string, module Module, registry *prometheus.Registry) (suc
...
+	if err != nil {
+		log.Errorf("Error generating TLS config: %v", err)
+	}
+
 	dial := func(network, address string) (net.Conn, error) {
 		return net.Dial(dialProtocol, address)
 	}
 	client.Transport = &http.Transport{
```
Comment (brian-brazil) [inline http.go]: ditto

**Comment by Conorbro** [comment]
@brian-brazil

File: http.go [hunk 4]
```diff
@@ -114,30 +113,13 @@ func probeHTTP(target string, module Module, registry *prometheus.Registry) (suc
...
-	}
-	dial := func(network, address string) (net.Conn, error) {
-		return net.Dial(dialProtocol, address)
-	}
-	client.Transport = &http.Transport{
-		TLSClientConfig:   tlsconfig,
-		Dial:              dial,
-		Proxy:             http.ProxyFromEnvironment,
-		DisableKeepAlives: true,
+		log.Errorf("Error generating HTTP client: %v", err)
```
Comment (brian-brazil) [inline http.go]: this should still return false

**Comment by Conorbro** [comment]
@brian-brazil

File: README.md [hunk 5]
```diff
@@ -54,6 +54,17 @@ modules:
...
         Content-Type: application/json
       body: '{}'
+  http_basic_auth_example:
+    prober: http
+    timeout: 5s
+    http:
+      method: POST
+      headers:
+        Host: "login.example.com"
+      http_client_config:
```
Comment (brian-brazil) [inline README.md]: This shouldn't be here, basic auth should be under http directly

**Comment by Conorbro** [comment]
@brian-brazil

**Pull request merged**